server/src/main/java/org/elasticsearch/common/util/CancellableSingleObjectCache.java

```diff
@@ -100,6 +100,13 @@ protected boolean isFresh(Key currentKey, Key newKey) {
         return currentKey.equals(newKey);
     }
 
+    /**
+     * Sets the currently cached item reference to {@code null}, which will result in a {@code refresh()} on the next {@code get()} call.
+     */
+    protected void clearCurrentCachedItem() {
```

Comment:
This isn't covered by the `CancellableSingleObjectCache` test suite yet. Also should this be `final`? We don't want subclasses overriding it.